lib/base/networking/apis/auth_handler.dart

```diff
@@ -0,0 +1,98 @@
+import 'package:cookie_jar/cookie_jar.dart';
+import 'package:dio/dio.dart';
+import 'package:dio_cookie_manager/dio_cookie_manager.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_inappwebview/flutter_inappwebview.dart' as webview;
+import 'package:gocast_mobile/base/helpers/model_generator.dart';
+import 'package:gocast_mobile/model/token_model.dart';
+import 'package:gocast_mobile/model/user/user_model.dart';
+import 'package:gocast_mobile/routes.dart';
+
+class AuthHandler {
+  static Future<void> basicAuth(
+    String username,
+    String password,
+  ) async {
+    const url = Routes.basicLogin;
+    final cookieJar = CookieJar();
+    final dio = Dio(
+      BaseOptions(
+        followRedirects: false,
+        validateStatus: (status) {
+          return status! < 500;
+        },
+      ),
+    );
+    dio.interceptors.add(CookieManager(cookieJar));
+
+    final formData = FormData.fromMap({
+      'username': username,
+      'password': password,
+    });
+
+    try {
+      await dio.post(
+        url,
+        data: formData,
+      );
+    } catch (e) {
+      // Throw the error so it can be caught and handled in the signIn method
```

Comment:
```suggestion
      // Throw the error so it can be caught and handled in the signIn method
      // Log the error
      debugPrint('Request error: $e');
```

Reply:
Will implement these changes when solving [this issue](https://github.com/TUM-Dev/gocast_mobile/issues/32).